lib/internal/streams/end-of-stream.js

```diff
@@ -7,6 +7,7 @@ const {
   ERR_INVALID_ARG_TYPE,
   ERR_STREAM_PREMATURE_CLOSE
 } = require('internal/errors').codes;
+const { kClosed } = require('internal/http');
```

Comment:
This is not correct. stream should not require http. It would also make implementing this in `readable-stream` impossible.

Reply:
@mcollina: The only alternatives as far as I see it then is to make `kClosed` a public property e.g. `_closed` or not fixing this. Any preference?

Reply:
I think we can use `_closed` as long as there is no writable or readable state attached to the stream.